### PR TITLE
#753 add the 'hideOnChildrenHidden' component param

### PR DESCRIPTION
**Pull request by kamsi**
Addresses: https://github.com/formio/formio.js/issues/753

### CODE REVIEW

File: src/components/columns/Column.js [hunk 1]
```diff
@@ -4,13 +4,14 @@ export default class ColumnComponent extends NestedComponent {
   constructor(component, options, data) {
     super(component, options, data);
     this.noEdit = true;
+    this.hideOnChildrenHidden = this.component.hideOnChildrenHidden != null ? this.component.hideOnChildrenHidden : true;
```
Comment (travist) [inline src/components/columns/Column.js]: We cannot change the default behavior. You are defaulting this to ```true``` when it should be ```false```. Because of this, I think the variable should be called ```showOnChildrenHidden``` and default it to ```false```. 

There is also a better way of doing this where you would just add this new property to the ```schema``` definition. Then you can put this in the check.

```js
if (allChildrenHidden && !this.component.showOnChildrenHidden) {

}
```

**Comment by kamsi** [comment]
@travist

My default is fine. It works exactly as it used to when you do not setup it: defaults to true and so the previous behavior (which I consider weird) is preserved and the component will hide. Please read it again carefully.

I also disagree with the rename to showOnChildrenHidden. What would it mean? That no matter anything else the component will show if it does not have children? I believe my name better reflects what is being done here: you will hide the component if hideOnCh... is true, but if it is not yoi will continue the regular resolution logic.

I could probably not write to component variable and resolve it straight from component... not sure that is such a gain though.

To sum it up

1. Code works fine as it is
2. Rename to shoOn... I consider semantically inapropriate because it will be misleading.

Yet, if you still think it should be showOn... tell me so and I will do it.

**Comment by travist** [comment]
I would normally like ```hideOnChildrenHidden```, but the issue I have has more to do with "default" behavior if that property is not present. What you have here suggests that if it is not present, it is TRUE since that is the current behavior. ***I typically don't like introducing variables that are the inverse of the default.***  ```showOnChildrenHidden``` does have meaning and is the inverse of the variable above. It also makes sense that it is FALSE if the variable is not present meaning that FALSE would work the same as if it were provided and false. This keeps the variable inline with "reverse compatibility".

I would say let's have another developer chime in and give their opinion on this. @randallknutson would be a good person to give his opinions.

**Comment by kamsi** [comment]
@travist sure, I will settle for whatever verdict @randallknutson gives.

I get your point on inverse. To be perfectly honest I believe that current default is not sane (I described that in the issue: https://github.com/formio/formio.js/issues/753 ). At least not for the empty case. Should we really hide component just because it does not have children?
My dreams come true would be actually a different default for the no children case.
So sth like:

  `
if (allChildrenHidden && this.getComponents().length) {
  return false;
}
`

But this would indeed break the current behavior for this edge case and hence, would probably have to be in major.

**Comment by randallknutson** [comment]
Ideally we would have made an option to hide a column if all components within it were hidden however we are now in a position where the default is to hide the column if the all components are hidden. 

For now the best we can do is to set the property in a backwards compatible way like Travis said. Moving forward this is something we should clean up in a new schema major version.

**Comment by travist** [comment]
@kamsi @randallknutson if I remember correctly, we recently added the functionality of hiding the columns if all children are hidden. I know it may sound blasphemous, but in this particular case, I am not going to put up much of a fight to keep reverse compatibility since it wasn't that many versions ago that we actually changed "default" behavior to hide the column when children are hidden. It should also be noted that we have always anchored to the way the Angular 1x renderer works when deciding default behavior for this renderer. The Angular 1.x renderer does not hide the column either.

We could say that move introduced a "bug" and are now creating a configuration so that it goes back to the way it was, but does allow you to make it hide to appease the customers we added this functionality for. Give us some time to reach out to these customers and get their take on how disruptive this would be for them, and then we should make it behave the way we think it should behave, regardless of keeping reverse compatibility in place.

**Comment by travist** [comment]
We will take this PR, and increment the minor version. The reasoning is that we recently broke reverse compatibility when we introduced the hide column on children hidden logic, so that could be considered an introduced bug. This will make it configurable and default it to show the column.

**Comment by kamsi** [comment]
@travist I think there is still some confusion here. My PR will make it so that the default behavior will hide that column when no children!
If you want to show the column by default when children are hidden, then hideOnCh... needs to be false.

**Comment by travist** [comment]
I made changes to your pull request to apply the correct default. Please verify on your end, but it should be good.

**Comment by kamsi** [comment]
@travist all good! Thank you.
